src/physics/grid/TwodSubgridCalculator.i.hh

```diff
@@ -0,0 +1,66 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file TwodSubgridCalculator.i.hh
+//---------------------------------------------------------------------------//
+#include "NonuniformGrid.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with defaults.
+ */
+CELER_FUNCTION
+TwodSubgridCalculator::TwodSubgridCalculator(const TwodGridData& grids,
+                                             const Values&       storage,
+                                             InterpT             x_loc)
+    : grids_{grids}, storage_(storage), x_loc_(x_loc)
+{
+    CELER_EXPECT(grids);
+    CELER_EXPECT(grids.values.back() < storage.size());
+    CELER_EXPECT(x_loc.index + 1 < grids.x.size());
+    CELER_EXPECT(x_loc.fraction >= 0 && x_loc_.fraction <= 1);
```

Comment:
`x_loc_.fraction < 1`?

Reply:
Originally it was `<`, but I figured that interpolation on the last point would be easy to do by constructing with `<=`. Currently though it's always `<` though, so I'll change it to the more restrictive setting and add a note that `==1` should also be OK.